lib/tapioca/compilers/dsl/active_support_current_attributes.rb

```diff
@@ -0,0 +1,120 @@
+      #
+      # this generator will produce an RBI file with the following content:
+      # ~~~rbi
+      # # typed: strong
+      #
+      # class Current
+      #   sig { returns(T.untyped) }
+      #   def self.account; end
+      #
+      #   sig { returns(T.untyped) }
+      #   def account; end
+      #
+      #   sig { params(account: T.untyped).returns(T.untyped) }
+      #   def self.account=(account); end
+      #
+      #   sig { params(account: T.untyped).returns(T.untyped) }
+      #   def account=(account); end
+      #
+      #   sig { params(user_id: Integer).void }
+      #   def self.authenticate(user_id); end
+      #
+      #   sig { returns(T.untyped) }
+      #   def self.helper; end
+      # end
+      # ~~~
+      class ActiveSupportCurrentAttributes < Base
+        extend T::Sig
+
+        sig do
+          override
+            .params(
+              root: Parlour::RbiGenerator::Namespace,
+              constant: T.class_of(::ActiveSupport::CurrentAttributes)
+            )
+            .void
+        end
+        def decorate(root, constant)
+          dynamic_methods = dynamic_methods_for(constant)
+          instance_methods = instance_methods_for(constant) - dynamic_methods
+          return if dynamic_methods.empty? && instance_methods.empty?
+
+          root.path(constant) do |k|
+            dynamic_methods.each do |method|
+              method = method.to_s
+              # We want to generate each method both on the class
+              generate_method(k, method, class_method: true)
+              # and on the instance
+              generate_method(k, method, class_method: false)
+            end
+
+            instance_methods.each do |method|
+              # instance methods are only elevated to class methods
+              # no need to add separate instance methods for them
+              method = constant.instance_method(method)
+              create_method_from_def(k, method, class_method: true)
+            end
+          end
+        end
+
+        sig { override.returns(T::Enumerable[Module]) }
+        def gather_constants
+          ::ActiveSupport::CurrentAttributes.descendants
+        end
+
+        private
+
+        sig { params(constant: T.class_of(::ActiveSupport::CurrentAttributes)).returns(T::Array[Symbol]) }
+        def dynamic_methods_for(constant)
+          constant.instance_variable_get(:@generated_attribute_methods)&.instance_methods(false) || []
+        end
+
+        sig { params(constant: T.class_of(::ActiveSupport::CurrentAttributes)).returns(T::Array[Symbol]) }
+        def instance_methods_for(constant)
+          constant.instance_methods(false)
+        end
+
+        sig { params(klass: Parlour::RbiGenerator::Namespace, method: String, class_method: T::Boolean).void }
+        def generate_method(klass, method, class_method:)
+          if method.end_with?("=")
+            parameter = Parlour::RbiGenerator::Parameter.new(T.must(method[0...-1]), type: "T.untyped")
```

Comment:
Added a `T.must` here since `Parameter#new` takes on a `String` and not a `T.nilable(String)`. Want to make sure this is the best solution.

Reply:
Weird that `method[0...-1]` would be considered nilable. 🤔

Reply:
Oh, I guess it is because `"abc"[10...1] == nil`. Actually, we don't have to worry about the parameter name since it does not matter. Let's just supply it as `"value"`.